jericho/jericho.py

```diff
@@ -369,36 +375,35 @@ def __init__(self, story_file, seed=-1):
         self.seed = seed
         self.frotz_lib.setup(self.story_file, self.seed)
         self.player_obj_num = self.frotz_lib.get_self_object_num()
+        md5hash = hashlib.md5(open(self.story_file, 'rb').read()).hexdigest()
+        self.bindings = _load_bindings(md5hash)
+        self.act_gen = TemplateActionGenerator(self.bindings) if self.bindings else None
 
     def __del__(self):
         if hasattr(self, 'frotz_lib'):
             self.frotz_lib.shutdown()
             dlclose_func(self.frotz_lib._handle)
 
-    def get_dictionary(self):
-        ''' Returns a list of :class:`jericho.DictionaryWord` words recognized\
-        by the game's parser. See :doc:`dictionary`. '''
-        word_count = self.frotz_lib.get_dictionary_word_count(self.story_file)
-        words = (DictionaryWord * word_count)()
-        self.frotz_lib.get_dictionary(words, word_count)
-        self.frotz_lib.ztools_cleanup()
-        return list(words)
-
-    def disassemble_game(self):
-        ''' Prints the subroutines and strings used by the game. '''
-        self.frotz_lib.disassemble(self.story_file)
-
-    def victory(self):
-        ''' Returns `True` if the game is over and the player has won. '''
-        return self.frotz_lib.victory() > 0
+    def reset(self, use_walkthrough_seed=False):
```

Comment:
Suggestion: maybe this class should have a `seed(seed, use_walkthrough_seed=False)` method (similar to Gym API) instead of having `use_walkthrough_seed` in `reset()`. By default, if the user doesn't call `.seed` the "walkthrough_seed" should be used if available.

Reply:
The difficulty with having a seed method is that it's only really possible to seed when you're calling frotz_lib.setup() (e.g. when the FrotzEnv is being created/reset).

I agree that the use_walkthrough_seed in reset is not great...

Reply:
I think something like this could work (need to adjust the default value for seed and its docstring)
```python
    def __init__(self, story_file, seed=-1):
        if not os.path.isfile(story_file):
            raise FileNotFoundError(story_file)

        self.frotz_lib = _load_frotz_lib()
        self.is_fully_supported = bool(self.frotz_lib.is_supported(story_file.encode('utf-8')))
        if not self.is_fully_supported:
            msg = ("Game '{}' is not fully supported. Score, move, change"
                   " detection will be disabled.").format(story_file)
            warnings.warn(msg, UnsupportedGameWarning)

        self.story_file = story_file.encode('utf-8')
        self.player_obj_num = self.frotz_lib.get_self_object_num()
        md5hash = hashlib.md5(open(self.story_file, 'rb').read()).hexdigest()
        self.bindings = _load_bindings(md5hash)
        self.act_gen = TemplateActionGenerator(self.bindings) if self.bindings else None
        self.seed(seed)

    def seed(self, seed=None):
        self.frotz_lib.shutdown()  # In case it is running.
        seed = seed or self.bindings.get('seed', -1)
        self._seed = seed
        self.frotz_lib.setup(self.story_file, self.seed)
```